provisioning/baremetal_pod.go

```diff
@@ -516,3 +517,32 @@ func ApplyNewOrUpdatedDeployment(client appsclientv1.DeploymentsGetter, targetNa
 
 	return nil
 }
+
+func getDeploymentCondition(deployment *appsv1.Deployment) appsv1.DeploymentConditionType {
+	for _, cond := range deployment.Status.Conditions {
+		if cond.Status == corev1.ConditionTrue {
+			return cond.Type
+		}
+	}
+	return appsv1.DeploymentProgressing
+}
+
+// Provide the current state of metal3 deployment
+func GetDeploymentState(client appsclientv1.DeploymentsGetter, targetNamespace string, config *metal3iov1alpha1.Provisioning) (appsv1.DeploymentConditionType, error) {
+	existing, err := client.Deployments(targetNamespace).Get(context.Background(), BaremetalDeploymentName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		// Deployment is not available and if Provisioning CR is not available,
+		// then everything is fine
+		if config == nil {
```

Comment:
This can't happen, so I wouldn't worry about it.

Reply:
It is possible that the provisioning CR was deleted and and as a response to that we deleted the metal3 deployment.

Reply:
Agreed. Fixed logic around this.